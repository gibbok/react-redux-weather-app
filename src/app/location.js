// @flow
import {LocationType} from '../types'  // eslint-disable-line no-unused-vars

const location = ():LocationType => {
  return {
    locationId: 5128581, // nyc usa
    geo: [-74.006, 40.7143],
    isFetching: false
  }
}

export default location
