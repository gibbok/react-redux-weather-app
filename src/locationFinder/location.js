const Location = ({onClick, name, country}) => (
  <li onClick={onClick}>
    {name} {country}
  </li>
)

export default Location
