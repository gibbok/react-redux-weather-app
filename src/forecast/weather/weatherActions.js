import 'isomorphic-fetch'
import * as api from '../../app/api'
import * as types from './weatherActionTypes'

const getWeather = query => ({
  type: types.GET_WEATHER,
  payload: new Promise((resolve, reject) => {
    fetch(api.weather(query)).then(response => {
      resolve(response.json())
    })
  })
})

export { getWeather }
