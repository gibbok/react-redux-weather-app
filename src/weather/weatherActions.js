import 'isomorphic-fetch'
import * as api from '../app/api'
import * as types from './weatherActionTypes'

const getLocation = query => ({
  type: types.GET_LOCATION,
  payload: new Promise((resolve, reject) => {
    fetch(api.weather(query)).then(response => {
      resolve(response.json())
    })
  })
})

export { getLocation }
