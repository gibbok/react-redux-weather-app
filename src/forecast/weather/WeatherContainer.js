import { connect } from 'react-redux'
import Weather from './Weather'
import { getWeather } from './weatherActions'
import store from '../../app/store'

let isInitiated = false

const weather = (weather) => {
  return weather
}

const getWeatherLocationId = () => {
  return store.getState().weatherReducer.weather.app.locationId
}

const onInit = () => {
  if (isInitiated) {
    return
  }
  isInitiated = true
  store.dispatch(getWeather(getWeatherLocationId()))
}

const mapStateToProps = (state) => {
  return {
    weather: weather(state.weatherReducer.weather.data),
    onInit: onInit()
  }
}

const WeatherContainer = connect(
  mapStateToProps
)(Weather)

export default WeatherContainer
