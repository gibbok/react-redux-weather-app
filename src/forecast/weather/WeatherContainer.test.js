import React from 'react'
import { createMockStore, createMockDispatch } from 'redux-test-utils'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import configureStore from 'redux-mock-store'
import WeatherContainer from './WeatherContainer'
import {getWeather} from './weatherActions'
import location from '../../app/location'

describe('WeatherContainer', () => {
  it('should render', () => {
    const initialState = {
      weatherReducer: {
        weather: {
          data: {},
          app: {
            locationId: location().locationId, // nyc usa
            isFetching: false
          },
          ui: {}
        }}
    }
    const mockStore = configureStore()
    const store = mockStore(initialState)
    const wrapper = shallow(<WeatherContainer store={store} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('should dispatch action getWeather', () => {
    const state = 'state'
    const store = createMockStore(state)
    const action = getWeather(location().locationId)

    store.dispatch(action)
    expect(store.getAction(action.type)).toEqual(action)
    expect(store.getActions()).toEqual([action])
    expect(store.isActionDispatched(action)).toBe(true)
    expect(store.isActionTypeDispatched(action.type)).toBe(true)
    expect(store.getState()).toBe(state)

    const dispatchMock = createMockDispatch()
    dispatchMock.dispatch(action)
    expect(dispatchMock.getAction(action.type)).toEqual(action)
    expect(dispatchMock.getActions()).toEqual([action])
    expect(dispatchMock.isActionDispatched(action)).toBe(true)
    expect(dispatchMock.isActionTypeDispatched(action.type)).toBe(true)
  })
})
