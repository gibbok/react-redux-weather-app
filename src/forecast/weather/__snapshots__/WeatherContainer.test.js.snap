// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherContainer should render 1`] = `
<withStyles(Weather)
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  weather={Object {}}
/>
`;
